Parsimony/ParserBuilder/PropertySelector.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Parsimony.ParserBuilder
+{
+    internal class PropertySelector<TInput, TProp>
+    {
+        /// <summary>
+        /// The name of the member the selector function targets
+        /// </summary>
+        public string MemberName { get; }
+
+        /// <summary>
+        /// A setter for the property 
+        /// </summary>
+        public Action<TInput, TProp> Setter { get; }
+
+        /// <summary>
+        /// A getter for the property
+        /// </summary>
+        public Func<TInput, TProp> Getter { get; }
+
+        /// <summary>
+        /// Constructs a new PropertySelector with the given selector function
+        /// </summary>
+        /// <exception cref="ArgumentNullException"> <paramref name="selector"/> is null </exception>
+        /// <exception cref="ArgumentException"> <paramref name="selector"/> is not of type <see cref="MemberExpression"/> </exception>
+        /// <exception cref="ArgumentException"> The member selected by <paramref name="selector"/> is not a property </exception>
+        /// <exception cref="ArgumentException"> The member selected by <paramref name="selector"/> is not readable and writable </exception>
+        /// <param name="selector"> A property selector function</param>
+        public PropertySelector(Expression<Func<TInput, TProp>> selector)
+        {
+            if (selector == null)
+                throw new ArgumentNullException(nameof(selector));
+
+            if (!(selector.Body is MemberExpression mbr))
+                throw new ArgumentException("Must be a member selector expression");
+
+            //TODO:CN -- Handle convert nodes? casts etc?
```

Comment:
I think the meaning of the expression would be sufficiently vague that they'd be better off putting logic in their property than doing something here. For example, if we cast an `int` property to `long` to satisfy the TValue, how do we know what to do with the `long` value that we parse to get it back into the `int` property?

Reply:
Fixed er